activerecord/lib/active_record/type/internal/timezone.rb

```diff
@@ -16,6 +16,10 @@ def is_utc?
         def default_timezone
           @timezone || ActiveRecord.default_timezone
         end
+
+        def ==(other)
+          super(other) && default_timezone == other.default_timezone
```

Comment:
I think we should consider `default_timezone` may change later, hence strictly compare `@timezone` not `default_timezone`.

We can make a `protected attr_reader :timezone` for that.